### ISSUE TITLE
Wrong scheme in torrent download link when behind reverse proxy

### ISSUE THREAD

**Issue by thexa4** [Jackett/Jackett#1464]
This can occur when using ssl termination using nginx as a reverse proxy. The download link in the torznab api endpoint returns a link that starts with http://

This results in a problem for applications that don't follow redirects correctly and leaks information when accessing jackett from an untrusted network.

This behavior originates in TorznabController.cs:150, it uses the scheme from the RequestUri which is always http.

Ideally one of these solutions would be added:
- Add support for the [Forwarded header](https://tools.ietf.org/html/rfc7239#section-4)
- Add support for the [X-Forwarded-Proto header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Proto)
- Add a configuration option to use a different scheme.

Jackett version:  0.7.1483.0
Mono version (if not using Windows):  Debian 4.6.2.7+dfsg-1

**Comment by DivinityCycle** [comment]
+1, encountered this last night. This effectively breaks manual search functionality for me. Good on your for tracing the issue to it's source!

**Comment by turtlehead** [comment]
I am also having this problem behind an apache reverse proxy.

**Comment by kaso17** [comment]
fixed via 3790cbb8944af3ae793726ec956ad1751d5604b9

An apache config like this should work:
```
ProxyPreserveHost On
RequestHeader set X-Forwarded-Proto "https"
ProxyPass /jackett http://127.0.0.1:9117/
```
